x/lockup/keeper/lock_test.go

```diff
@@ -96,143 +65,156 @@ func (suite *KeeperTestSuite) TestGetPeriodLocks() {
+			expectedBeginUnlockPass:       false,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+		{
+			name:                          "partial unlocking",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("stake", 5)},
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            sdk.Coins{sdk.NewInt64Coin("stake", 5)},
+		},
+		{
+			name:                          "partial unlocking unknown tokens",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("unknown", 5)},
+			expectedBeginUnlockPass:       false,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+		{
+			name:                          "unlocking should not finish yet",
+			unlockingCoins:                initialLockCoins,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Millisecond,
+			expectedUnlockMaturedLockPass: false,
+			balanceAfterUnlock:            sdk.Coins{},
+		},
+	}
 
-	addr1 := sdk.AccAddress([]byte("addr1---------------"))
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	for _, tc := range testCases {
+		suite.SetupTest()
 
-	// lock with balance
-	suite.FundAcc(addr1, coins)
-	lock, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, addr1, coins, time.Second)
-	suite.Require().NoError(err)
+		addr1 := sdk.AccAddress([]byte("addr1---------------"))
+		lock := types.NewPeriodLock(1, addr1, time.Second, time.Time{}, initialLockCoins)
 
-	// begin unlock with lock object
-	err = suite.App.LockupKeeper.BeginUnlock(suite.Ctx, lock.ID, nil)
-	suite.Require().NoError(err)
+		// lock with balance
+		suite.FundAcc(addr1, initialLockCoins)
+		lock, err := suite.App.LockupKeeper.CreateLock(suite.Ctx, addr1, initialLockCoins, time.Second)
+		suite.Require().NoError(err)
 
-	lockPtr, err := suite.App.LockupKeeper.GetLockByID(suite.Ctx, lock.ID)
-	suite.Require().NoError(err)
+		// store in variable if we're testing partial unlocking for future use
+		partialUnlocking := tc.unlockingCoins.IsAllLT(initialLockCoins) && tc.unlockingCoins != nil
 
-	// unlock with lock object
-	err = suite.App.LockupKeeper.UnlockMaturedLock(suite.Ctx.WithBlockTime(now.Add(time.Second)), lockPtr.ID)
-	suite.Require().NoError(err)
-}
+		// begin unlocking
+		err = suite.App.LockupKeeper.BeginUnlock(suite.Ctx, lock.ID, tc.unlockingCoins)
 
-func (suite *KeeperTestSuite) TestPartialUnlock() {
-	suite.SetupTest()
-	now := suite.Ctx.BlockTime()
+		if tc.expectedBeginUnlockPass {
+			suite.Require().NoError(err)
 
-	addr1 := sdk.AccAddress([]byte("addr1---------------"))
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+			// check unlocking coins
+			unlockingCoins := tc.unlockingCoins
+			if tc.unlockingCoins == nil {
+				unlockingCoins = initialLockCoins
```

Comment:
Suggesting to differentiate actual vs expected names more explicitly
```suggestion
			expectedUnlockingCoins := tc.unlockingCoins
			if expectedUnlockingCoins == nil {
				expectedUnlockingCoins = initialLockCoins
```